File: 프로그래머스/unrated/136798. 기사단원의 무기/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i = 1; i <= number; i++) {
            int count = isPrime(i);
            answer += (count > limit) ? power : count;
        }

        return answer;
    }

    public static int isPrime(int num) {
        int result = 0;

        for(int i = 1; i <= Math.sqrt(num); i++) {
            if(num % i == 0) {
                // i로 나눠지면 약수의 개수 2개 추가
                result += 2;

                if(num / i == i) {
                    // ex) num = 4일 때 i = 2면 result +=2 에서 -1 해준다.
                    result -= 1;
                }
            }
        }

        return result;
    }
}

File: 프로그래머스/1/136798. 기사단원의 무기/기사단원의 무기.py
def solution(number, limit, power):
    answer = 0
    a = []
    count = 0
    for i in range(1, number + 1) :
        for j in range(1, int(i ** (1/2) + 1)) :
            if (i % j == 0) :
                count += 1
                if (j != i // j) :
                    count += 1
    
        a.append(count)
        count = 0
    
    for i in range(len(a)) :
        if (a[i] > limit) :
            answer += power
        else :
            answer += a[i]
    
    return answer